pkg/signup/service/signup_service.go

```diff
@@ -56,6 +57,13 @@ func NewSignupService(context servicecontext.ServiceContext, cfg ServiceConfigur
 // newUserSignup generates a new UserSignup resource with the specified username and userID.
 // This resource then can be used to create a new UserSignup in the host cluster or to update the existing one.
 func (s *ServiceImpl) newUserSignup(ctx *gin.Context) (*v1alpha1.UserSignup, error) {
+	username := ctx.GetString(context.UsernameKey)
+
+	usersignup.TransformUsername(username)
```

Comment:
you need to use the output of the transformation: 
```suggestion
	username = usersignup.TransformUsername(username)
```

Reply:
woopsie, looks like i got too excited. fixed here: b49f7bb7673a86099e9ab0fce3101f46f391503f